app/code/Magento/QuoteGraphQl/Model/Cart/SetBillingAddressOnCart.php

```diff
@@ -101,6 +110,12 @@ private function createBillingAddress(
     ): Address {
         if (null === $customerAddressId) {
             $billingAddress = $this->quoteAddressFactory->createBasedOnInputData($addressInput);
+
+            $customerId = $context->getUserId();
+            // need to save address only for registered user and if save_in_address_book = true
+            if (0 !== $customerId && !empty($addressInput['save_in_address_book'])) {
```

Comment:
Please throw an exception for guests.

Reply:
Done.

Reply:
I believe we can move this code to the top of the method.
```
if (false === $context->getExtensionAttributes()->getIsCustomer()) {
    throw new GraphQlAuthorizationException(__('The current customer isn\'t authorized.'));
}
```
This will simplify the complexity.
And if value is expected to be `boolean`, please use strict comparison.
```
(bool)$addressInput['save_in_address_book'] === false
```

Reply:
@lenaorobei, Seems like we can't as  `if (null === $customerAddressId)` we are creating billing address for guest.
Array may don't have  "save_in_address_book". So we will need double check: for isset and `(bool)$addressInput['save_in_address_book'] === true`. Can't we just leave only check for `empty()`?

Reply:
Because of PHP type casting please use `isset` + `=== true`.

Reply:
Done